ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ProcessingLogConfig.java

```diff
@@ -0,0 +1,131 @@
+  private static final String STREAM_NAME_DOC =
+      "If automatic processing log stream creation is enabled, KSQL sets the name of the "
+          + "stream to the value of this property.";
+
+  public static final String TOPIC_NAME = propertyName("topic.name");
+  public static final String TOPIC_NAME_NOT_SET = "";
+  public static final String TOPIC_NAME_DEFAULT_SUFFIX = "ksql_processing_log";
+  private static final String TOPIC_NAME_DOC =
+      "If automatic processing log topic creation is enabled, KSQL sets the name of the "
+          + "topic to the value of this property. If automatic processing log stream "
+          + "creation is enabled, KSQL uses this topic to back the stream.";
+
+  public static final String TOPIC_PARTITIONS = propertyName("topic.partitions");
+  private static final int TOPIC_PARTITIONS_DEFAULT = 1;
+  private static final String TOPIC_PARTITIONS_DOC =
+      "If automatic processing log topic creation is enabled, KSQL creates the topic with "
+          + "number of partitions set to the value of this property.";
+
+  public static final String TOPIC_REPLICATION_FACTOR = propertyName("topic.replication.factor");
+  private static final short TOPIC_REPLICATION_FACTOR_DEFAULT = 1;
+  private static final String TOPIC_REPLICATION_FACTOR_DOC =
+      "If automatic processing log topic creation is enabled, KSQL creates the topic with "
+          + "number of replicas set to the value of this property.";
+
+  public static final String STREAM_AUTO_CREATE = propertyName("stream.auto.create");
+  private static final String STREAM_AUTO_CREATE_DOC = String.format(
+      "Toggles automatic processing log stream creation. If set to \"%s\", and "
+          + "running interactive mode on a new cluster, then KSQL will automatically "
+          + "create a processing log stream when it starts up. The name for the stream "
+          + "is the value of the \"%s\" property. The stream will be created over the topic "
+          + "set in the \"%s\" property",
+      AUTO_CREATE_ON,
+      STREAM_NAME,
+      TOPIC_NAME);
+
+  public static final String TOPIC_AUTO_CREATE = propertyName("topic.auto.create");
+  private static final String TOPIC_AUTO_CREATE_DOC = String.format(
+      "Toggles automatic processing log topic creation. If set to \"%s\", then "
+          + "KSQL will automatically try to create a processing log topic at startup. "
+          + "The name of the topic is the value of the \"%s\" property. The number of "
+          + "partitions is taken from the \"%s\" property , and the replication factor "
+          + "is taken from the \"%s\" property",
+      AUTO_CREATE_ON,
+      TOPIC_NAME,
+      TOPIC_PARTITIONS,
+      TOPIC_REPLICATION_FACTOR);
+
+  private static final Validator AUTO_CREATE_VALIDATOR
+      = ValidString.in(AUTO_CREATE_ON, AUTO_CREATE_OFF);
+
+  private static final ConfigDef CONFIG_DEF = new ConfigDef()
+      .define(
+          STREAM_AUTO_CREATE,
```

Comment:
Why not define this as a BOOLEAN type, rather than having a string value that only accepts ON or OFF??

Reply:
I generally prefer not to use booleans, since with the other types you always have some wiggle room to extend the behavior if you need to.

Reply:
I hear you, but this is planning of a future enhancement that is unlikely to happen.  We have a boolean type.  In all likelihood on/off is all we will every need.  If, and it's a big if, we need a third value, then we can migrate to a new config.  Yes that's a pain, but having a boolean config that requires users to know the correct 'on' / 'off', or 'true' / 'false', (you're code is also case-sensitive), well, that justs a PITA for our users whenever they're using this.

I'd really urge you to go with a boolean if you're no definitely plans to extend this beyond on/off.

Reply:
Yeah I don't have a great argument beyond the extensibility, which I agree is unlikely in this case. I'll switch it to a boolean.